src/scene/shader-lib/program-library.js

```diff
@@ -95,7 +95,7 @@ class ProgramLibrary {
 
             const device = this._device;
             def = generator.createShaderDefinition(device, options);
-            def.name = `${name}-pass:${options.pass}`;
+            def.name = `${name}-pass:${options.litOptions?.pass}`;
```

Comment:
I think .pass should stay on options as well to avoid changes in this file, as it needs to be compatible with other generators (for example basic.js) which has uses options.pass.

Reply:
The shader backend needs access to the pass, the front end doesn't. The backend only has the backend options, so in that case we need to duplicate the pass. How do we solve that?

Reply:
yeah duplication should be fine I think.